### PR TITLE
Improve multi homing  user documentation

### PR DESCRIPTION
**Pull request by maiqueb**
**- What this PR does and why is it needed**

This PR provides a detailed explanation of each available attribute for layer2/layer3 topologies.

It also clarifies that the secondary networks constructs are **not** namespaced.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: docs/multi-homing.md [hunk 1]
```diff
@@ -22,8 +22,11 @@ configurations are defined in a CRD named `NetworkAttachmentDefinition`.
 Below you will find example attachment configurations for each of the current
 topologies OVN-K allows for secondary networks.
 
-**NOTE**: currently, all the secondary networks **only** allow for east/west
-traffic.
+**NOTE**:
+- currently, all the secondary networks **only** allow for east/west traffic.
+- these networks are **not** namespaced - i.e. you cannot create multiple
+  `network-attachment-definition`s pointing at the same network (i.e. sharing
+  the `name` attribute) with different configurations in different namespaces.
```
Comment (jcaamano) [inline docs/multi-homing.md:29]: nit: I would rephrase:
```
- networks are **not** namespaced - i.e. creating multiple
  `network-attachment-definition`s with different configurations pointing at the same network (same
   NetConf `name` value) is not supported.
```

File: docs/multi-homing.md [hunk 2]
```diff
@@ -22,8 +22,11 @@ configurations are defined in a CRD named `NetworkAttachmentDefinition`.
 Below you will find example attachment configurations for each of the current
 topologies OVN-K allows for secondary networks.
 
-**NOTE**: currently, all the secondary networks **only** allow for east/west
-traffic.
+**NOTE**:
+- currently, all the secondary networks **only** allow for east/west traffic.
```
Comment (jcaamano) [inline docs/multi-homing.md:26]: nit: I would have this comment on each network topology section as this would no longer hold true when localnet is added

Comment (maiqueb) [reply docs/multi-homing.md:26]: IMO, either localnet lands first, and this has to be re-written, or this lands first, and localnet has to update the README. Something they need to do any way, right ?

Comment (jcaamano) [reply docs/multi-homing.md:26]: You can always do something so that it is easier or more difficult to update in the future and we already know what's coming. But fair enough.

File: docs/multi-homing.md (hunk 1 above)
Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md (hunk 2 above)
Comment (maiqueb) [reply docs/multi-homing.md:26]: OK, resolving.

File: docs/multi-homing.md [hunk 3]
```diff
@@ -22,8 +22,12 @@ configurations are defined in a CRD named `NetworkAttachmentDefinition`.
...
+- networks are **not** namespaced - i.e. creating multiple
+  `network-attachment-definition`s with different configurations pointing at the
+  same network (same `NetConf.Name` attribute) with different configurations in
+  different namespaces is **not** supported.
```
Comment (jcaamano) [inline docs/multi-homing.md:30]: `with different configurations` is said twice.
Also wanted to avoid `in different namespaces` because is not really relevant, in is not allowed in different namespaces or in same namespace.

Comment (maiqueb) [reply docs/multi-homing.md]: Done.

File: docs/multi-homing.md [hunk 4]
```diff
@@ -88,6 +97,24 @@ spec:
...
+#### Network Configuration reference
+- `name` (string, required): the name of the network. This attribute is **not** namespaced.
+- `type` (string, required): "ovn-k8s-cni-overlay".
+- `topology` (string, required): "layer2".
+  `subnets` (string, optional): a comma separated list of subnets. When multiple subnets
+  are provided, the user will get an IP from each subnet.
+- `mtu` (integer, optional): explicitly set MTU to the specified value. Defaults to the value chosen by the kernel.
+- `netAttachDefName` (string, required): must match <namespace>/<net-attach-def name>
+  of the surrounding object.
+- `excludeSubnets` (string, optional): a comma separated list of CIDRs / IPs.
```
Comment (weliang1) [inline docs/multi-homing.md:109]: Could you have a sample of excludeSubnets configuration in NetworkAttachmentDefinition?

Comment (maiqueb) [reply docs/multi-homing.md:110]: Done.
